launch/launch/launcher.py

```diff
@@ -0,0 +1,154 @@
+import asyncio
+import signal
+from threading import Lock
+
+from launch import ProcessDescriptor
+from launch.output_handler import CompositeOutputHandler
+from launch.output_handler import ConsoleOutput
+from launch.exit_handler import DefaultExitHandler
+from launch.protocol import SubprocessProtocol
+
+
+class DefaultLauncher(object):
```

Comment:
We currently use it in 33 other places with the `object` and only in one location without. I would keep it for now but we can certainly remove it in all locations if everyone agrees.